cedar-policy-core/src/ast/name.rs

```diff
@@ -570,11 +570,11 @@ impl From<ReservedNameError> for ParseError {
     fn from(value: ReservedNameError) -> Self {
         ParseError::ToAST(ToASTError::new(
             value.clone().into(),
-            match &value.0.loc {
-                Some(loc) => loc.clone(),
+            match value.0.loc {
+                loc @ Some(_) => loc,
                 None => {
                     let name_str = value.0.to_string();
-                    Loc::new(0..(name_str.len()), name_str.into())
+                    Some(Loc::new(0..(name_str.len()), name_str.into()))
                 }
```

Comment:
if you want, instead of `match` this could be `value.0.loc.or_else(|| ...)`

Reply:
I did this then decided against it because then you need to compute `name_str` twice (since you're in the closure) or pre-compute it. Open to suggestions if you know how to make it clean.

Reply:
```
value.0.loc.or_else(|| {
  let name_str = value.0.to_string();
  Some(Loc::new(0..(name_str.len()), name_str.into()))
});
```